core/src/main/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbe.java

```diff
@@ -70,72 +68,79 @@ public class HasUnreleasedProductionChangesProbe  extends Probe {
     @Override
     public ProbeResult doApply(Plugin plugin, ProbeContext context) {
         List<String> productionPathsToCheckForCommits = new ArrayList<>();
-        Set<String> commitsAfterReleaseDate  = new HashSet<>();
+        Set<String> filesModifiedAfterRelease  = new HashSet<>();
 
         final Matcher matcher = SCMLinkValidationProbe.GH_PATTERN.matcher(plugin.getScm());
         if (!matcher.find()) {
             return ProbeResult.failure(key(), "The SCM link is not valid");
         }
         final String folder = matcher.group("folder");
 
-        productionPathsToCheckForCommits.add("pom.xml");
-        productionPathsToCheckForCommits.add(folder + "pom.xml");
-        productionPathsToCheckForCommits.add(folder + "src/main");
-
         try (Git git = Git.init().setDirectory(context.getScmRepository().toFile()).call()) {
-            final LogCommand logCommand = git.log().setMaxCount(1);
+            final LogCommand logCommand = git.log();
+
             if (folder != null) {
-                logCommand.addPath(folder);
+                productionPathsToCheckForCommits.add(folder + "pom.xml");
+                productionPathsToCheckForCommits.add(folder + "src/main");
             }
+
+            productionPathsToCheckForCommits.add("pom.xml");
+            productionPathsToCheckForCommits.add("src/main");
+
             productionPathsToCheckForCommits.forEach(logCommand :: addPath);
             Iterable<RevCommit> commits = logCommand.call();
 
-            for( RevCommit commit : commits ) {
+            for ( RevCommit commit : commits ) {
                 Instant instant = Instant.ofEpochSecond(commit.getCommitTime());
                 ZonedDateTime zonedDateTime = ZonedDateTime.ofInstant(instant, ZoneId.systemDefault());
 
-                if(commit.getParentCount() > 0 ) {
+                if (commit.getParentCount() > 0 ) {
                     /*
                     *  if a previous commit exists, compare the difference
                     * */
 
-                    // https://stackoverflow.com/a/27375013/9493145
-
+                    ObjectId oldCommit = git.getRepository().resolve("HEAD^");
```

Comment:
For the code that executes in `if` condition I get the following error:

```
java.lang.RuntimeException: org.eclipse.jgit.errors.IncorrectObjectTypeException: Object 6c53b498aa3e0785e46c11d0ee58bebfa1f4a4cf is not a tree.

```

Reply:
It says it is not a tree. 

The exception is thrown at `oldTreeIter.reset(reader, oldCommit);`

According to the method signature I do I am passing the correct objects.

Reply:
This is resolved.